.github/workflows/build.yml

```diff
@@ -0,0 +1,41 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  verify:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java-version: [ 8, 11 ]
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+      - name: Install Java and Maven
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java-version }}
+          distribution: 'temurin'
+      - run: mvn verify
+
+  deploy:
+    needs: verify
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+      # based on this : https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#yaml-example
+      - name: setup maven
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: SONATYPE_USERNAME
+          server-password: SONATYPE_PASSWORD
+
+      - name: publish to snapshot
+        run: mvn versions:set -DnewVersion=HEAD-SNAPSHOT clean deploy -B -X
```

Comment:
why not check in HEAD-SNAPSHOT if we think that is better for these builds? that way if you are building locally you get the same version string to work with?